Actions/AL-Go-Helper.ps1

```diff
@@ -661,6 +661,12 @@ function ReadSettings {
             "defaultReleaseMD"                          = "## Release reference documentation\n\nThis is the generated reference documentation for [{REPOSITORY}](https://github.com/{REPOSITORY}).\n\nYou can use the navigation bar at the top and the table of contents to the left to navigate your documentation.\n\nYou can change this content by creating/editing the **{INDEXTEMPLATERELATIVEPATH}** file in your repository or use the alDoc:defaultReleaseMD setting in your repository settings file (.github/AL-Go-Settings.json)\n\n{RELEASENOTES}"
         }
         "trustMicrosoftNuGetFeeds"                      = $true
+        "commitMessageSuffix"                           = ""
+        "commitOptions"                                 = [ordered]@{
+            "commitMessageSuffix"                       = ""
+            "commitAutoMerge"                           = $false
+            "commitPullRequestLabels"                   = @()
+        }
```

Comment:
Maybe add "alwaysUseGhTokenWorkflow" and "alwaysUseDirectCommit" in commitOptions - where false will use the input values and true will always use...

Reply:
We can add those options as well, but wouldn't you expect the input to override whatever settings we have defined? That's how the "Template Repository URL" input behaves isn't it? 

![image](https://github.com/user-attachments/assets/c6d5524b-1957-46a4-ab38-52b55c1a5378)

Reply:
I wouldn't "just" add these options - I would add an option called "alwaysUseDirectCommit"
If true - it would always use direct commit (also if the checkbox was left to false)
If false (or non existing) - it would use the value of the checkbox.

I wouldn't add the repository URL or the download latest to commitOptions - but I would add one called alwaysUseGhTokenWorkflow (which also is a checkbox and using the same rules as above).

Reply:
> If true - it would always use direct commit (also if the checkbox was left to false)

My point was... The above isn't consistent with how the "Template Repository URL" input works. If you define that input, we will override the templateUrl specified in settings. I think that's fine, but we should keep that consistent across AL-Go. IMO it should behave as if inputs are an 8th entry in this: https://github.com/microsoft/AL-Go/blob/main/Scenarios/settings.md#where-are-the-settings-located. 

I think if a user specifies one thing in the checkbox but something else happens that would be quite confusing. 

Thoughts?

Reply:
I kind of agree, but the problem with checkboxes is that they are not 3 state...
What does an empty checkbox mean? that it is defined as empty or that it is not set?
Which is why I would (for checkboxes) use the always- paradigm.
If the messagesuffix was an input field - the setting would be the default.

Reply:
If we want "Direct Commit" to have 3 states, it can't also be a checkbox since they only have two states. Checkboxes require the user to make a decision about whether they want the input to be true or false. You can't "leave it empty" - You're forced to make a decision. 

Maybe directCommit and useGhTokenWorkflow need to be "choice fields" that default to empty? 
![image](https://github.com/user-attachments/assets/9360e3d9-06cc-4611-8324-d9456e9c8b53)

> Which is why I would (for checkboxes) use the always- paradigm.

Always- paradigm being? If a setting has the "always" prefix it should take precedence even if a setting further down the chain is set to a different value?

Reply:
Yes - I am not talking about a generic option for all checkboxes - I am only talking about these two options:

DirectCommit is default false - if people allow direct commit (in github settings - i.e. no branch policies etc) and want to use that - it is something that would be fine to have globally - i.e. alwaysUseDirectCommit - then these workflows would always use direct commits and they wouldn't forget to check the checkbox.

Use GhTokenWorkflow is the same - if you have this secret defined and you want to use that - it would be fine to have a global setting to always use that. When people are not using GhTokenWorkflow - then the Create New App and other workflows will not spawn a CI/CD workflow - people who have setup the secrets and wants to use this don't want to forget to check the checkbox.

Reply:
> then these workflows would always use direct commits and they wouldn't forget to check the checkbox.

I still think this has the potential to be quite confusing (if you specify one thing in the checkbox but something else happens). Of course, we could say it's their own fault for using the alwaysUseDirectCommit setting but the user who triggers the workflow might not know that this setting is there and what it does. Or, they could've just forgotten or misunderstood how it works.

If it's okay with you, I'll park this as a future improvement. IMO I think we first need to figure out how we generally handle inputs vs settings in both the scheduled and manually triggered scenario.